### PR TITLE
Custom cookie loaders

### LINKED ISSUE #41: Custom cookie loading
I was assigned the task (by Bot Services) of making this dependency compatible with our hosting provider (Heroku). As stated in the README, "we store the cookie internally in a file, and files do not persist in Heroku." I plan to resolve this by storing the cookie in a database.

If I make a pull request to support this, which would be preferable (please add a reaction or comment)?

- **Flexibility**: I add an option to provide functions for saving and loading the cookie. ❤️ 
- **Ease-of-use**: I add an option to provide a connection url to a JDBC-compatible database (requires Java 1.7+). 🚀 
- **Both**: I could do both, whoever in that case, I'd be more inclined to provide *examples* for the save and load functions that are tailored to specific database types (MySQL, SQL Server, PostgreSQL). 👀

### LINKED ISSUE #48: Allow configuring cookie file location
A simple argument for where the cookie file is/what it's named would make deploying apps based on noblox easier.

### PR DESCRIPTION
**Pull request by cloewen8**
This change allows for the use of custom functions for loading and saving cookies by the relog utility.

`loadCookie(): ?{{string} cookie, {?number} time}`
`saveCookie(cookie: string): void`

It includes documentation in README.md and 3 sample loaders for popular databases (Postgres, Mysql and Msql).

# Notes
Automated tests have been run whoever I did not notice any unit tests. If all of the dynamic tests are manual, do I need to run or schedule them?

closes #41 
fixes #48

### CODE REVIEW

File: lib/user/cookieLogin.js [hunk 1]
```diff
@@ -8,11 +8,11 @@ var interval
 var day = 86400000
 exports.func = function (args) {
   // Run relog
-  return relog(args.cookie)
+  return relog(args)
     .then(function (r) {
       // Start interval
       if (!interval) {
-        setInterval(relog, day)
+        setInterval(() => relog(args), day)
```
[the remark this answers is not in the document]
Comment (cloewen8) [reply lib/user/cookieLogin.js]: Absolutely. I was unaware that you could pass arguments through setInterval. I'd way rather use standard features if possible.

**Comment by cloewen8** [comment]
At this point, I have finished everything I plan to add.

**Comment by cloewen8** [comment]
fixes #48

**Comment by gt-c** [comment]
What is preventing this from being merged?

**Comment by cloewen8** [comment]
I have experienced difficulty contacting the owner to assist in working on this framework. As far as I am aware no changes have been requested.

**Comment by suufi** [comment]
The only apprehension I have about this is the unreliability of Roblox's API. I haven't seen many requests for this feature either.

**Comment by chrissie1234** [comment]
I would recommend to merge.

**Comment by cloewen8** [comment]
I'm honestly surprised this hasn't been closed by now. I made this back in February 2019, 14 months ago. Early on a minor correction was requested but I never received any requests for changes after that.
I've since moved on from Roblox entirely (due to disagreements with Roblox's privacy policy) so I won't be making any updates to it beyond this simple merge conflict resolution today.

**Comment by AnthonyFuller** [comment]
This could be closed now thanks to v4.6.0

**Pull request closed without merging**